transcom/mymove
# B 20256-B Reapply int prevent prime users from accessing pptas

## [B-20256](Prevent Prime users from accessing PPTAS)

## Summary

Added Client cert authorization to PPTAS if the "allow_pptas" flag is set to true for a given user, and prevent acess if flag is false. Added pptas-client-api command

## How to test:

1.  you will need to get your CAC cert added to the client_cert table. Guide to do so here: [How to Create CAC Access](https://dp3.atlassian.net/wiki/spaces/MT/pages/1751089156/How+to+Create+CAC+Access+for+using+Prime+API+and+uploading+Electronic+Orders)
2. run make `server_run` and make `client_run`
3. log into [adminlocal](http://adminlocal:3000/sign-in)
4. go to [client_certs](http://adminlocal:3000/system/client-certificates)
5. find your cert. edit to set `allow_pptas` to true, then save
6. run the pptas-api-client command, using `go run ./cmd/pptas-api-client --cac --insecure --port 9443 list-moves --since '2024-01-01'` this should return some moves as a json.
7. switch `allow_pptas` to false
8. run `go run ./cmd/pptas-api-client --cac --insecure --port 9443 list-moves --since '2024-01-01'` again, a forbidden error should be returned
9. switch `allow_pptas` back to true
10. run `go run ./cmd/pptas-api-client --cac --insecure --port 9443 list-moves --since '2024-01-01'` again, this time moves should be returned as it has previously
